blockprotocol/blockprotocol
# [SITE] Fix the rendering for Embed Block in Hub

## What this Does

While the blocks available in blockprotocol are available for demo in [the hub](https://blockprotocol.org/hub), some of the blocks do not work as intended due to missing functions or data. This PR attempts to fix [the embed block](https://blockprotocol.org/@hash/embed) by providing the missing data and functions to it.

The changes include:
- Introducing block state and allowing it to be updated via the `updateEntities` function: [link](https://github.com/blockprotocol/blockprotocol/pull/116/files#diff-aa262447e474cd1fbf253d93d617df697f68d9ccf4af71e9a5e1aa766bd9cd19R38)
- Expanding upon the existing getEmbed function in utils: [link](https://github.com/blockprotocol/blockprotocol/pull/116/files#diff-61d9ccc6a7898cf7091df86378e0c98ba472e99b70b9a7bc3e02087689631736R95)
- Create an API to fetch oembed data from providers: [link](https://github.com/blockprotocol/blockprotocol/pull/116/files#diff-9cb3239510c47823de2d2af4fdb9a7686beda24bc0cc71d2a2cead277df06157R1)

## Follow Ups

- Get https://github.com/hashintel/hash/pull/211 merged.
- [Add support for `maxHeight` and `maxWidth` in resizer (for media + embed blocks)](https://app.asana.com/0/1201668052739245/1201720151462503/f)

## How to test it

1. Run the setup locally
2. Navigate to the embed block
3. Try pasting in different source URLs locally (eg, Twitter, YT, etc)

![chrome_aIESlI7by1](https://user-images.githubusercontent.com/22196279/150751142-14135a0d-acfb-44f6-8e6d-413c2bbb250a.png)

![chrome_a0rDnX0f7F](https://user-images.githubusercontent.com/22196279/150751149-0a86ee0b-a731-4b69-9c28-b976062bf0db.png)




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201697302996563
